MIGRATION.ru.md

```diff
@@ -1,5 +1,778 @@
+
+Было:
+
+```js
+modules.define('my-dom-block', ['i-bem__dom'], function(provide, BEMDOM) {
+
+provide(BEMDOM.decl(this.name, { /* ... */ }, {
+    live : true
+}));
+
+});
+```
+
+Стало:
+
+```js
+modules.define('my-dom-block', ['i-bem-dom'], function(provide, bemDom) {
+
+provide(bemDom.declBlock(this.name, { /* ... */ }, {
+    lazyInit : true
+}));
+
+});
+```
+
+Было:
+
+```js
+modules.define('my-dom-block', ['i-bem__dom'], function(provide, BEMDOM) {
+
+provide(BEMDOM.decl(this.name, { /* ... */ }, {
+    live : function() {
+        /* ... */
+    }
+}));
+
+});
+```
+
+Стало:
+
+```js
+modules.define('my-dom-block', ['i-bem-dom'], function(provide, bemDom) {
+
+provide(bemDom.declBlock(this.name, { /* ... */ }, {
+    lazyInit : true,
+
+    onInit : function() {
+        /* ... */
+    }
+}));
+
+});
+```
+
+Было:
+
+```js
+modules.define('my-dom-block', ['i-bem__dom'], function(provide, BEMDOM) {
+
+provide(BEMDOM.decl(this.name, { /* ... */ }, {
+    live : function() {
+        /* ... */
+        return false;
+    }
+}));
+
+});
+```
+
+Стало:
+
+```js
+modules.define('my-dom-block', ['i-bem-dom'], function(provide, bemDom) {
+
+provide(bemDom.declBlock(this.name, { /* ... */ }, {
+    onInit : function() {
+        /* ... */
+    }
+}));
+
+});
+```
+
+Было:
+
+```js
+{
+    block : 'b1',
+    js : { live : false }
+}
+```
+
+Стало:
+
+```js
+{
+    block : 'b1',
+    js : { lazyInit : false }
+}
+```
+
+Задача: [#877](https://github.com/bem/bem-core/issues/877).
+
+### Экземпляры для элементов
+
+Удалены элемент `elem-instances` блока `i-bem` и модификатор `elem-instances` элемента `dom` блока `i-bem`.
+Теперь соответсвующая функциональность является частью `i-bem` и `i-bem-dom`.
+
+Было:
+
+```js
+modules.define('my-dom-block__my-elem', ['i-bem__dom'], function(provide, BEMDOM) {
+
+provide(BEMDOM.decl({ block : 'my-dom-block', elem : 'my-elem' }, { /* ... */ }));
+
+});
+```
+
+Стало:
+
+```js
+modules.define('my-dom-block__my-elem', ['i-bem-dom'], function(provide, bemDom) {
+
+provide(bemDom.declElem('my-dom-block', 'my-elem', { /* ... */ }));
+
+});
+```
+
+Теперь метод `_elem()` экземпляра блока (бывший `elem()`) возвращает не jQuery-объект, а экземпляр класса элемента.
+
+#### Cпособы взаимодействия с элементами
+
+Было:
+
+```js
+this.setMod(this.elem('my-elem'), 'my-mod', 'my-val');
+```
+
+Стало:
+
+```js
+this._elem('my-elem').setMod('my-mod', 'my-val');
+```
+
+Аналогично для методов `getMod()`, `hasMod()`, `toggleMod()`, `delMod()`.
+
+#### Удалённые методы и поля
+
+Из API блока удачены методы: `elemify()` `elemParams()`, `getMods()` и поле `onElemSetMod`. Соответсвующая
+им функциональность выражается через экземпляры элементов.
+
+См. так же изменения про [методы поиска](#Методы-поиска).
+
+Задача: [#581](https://github.com/bem/bem-core/issues/581).
+
+### Методы поиска
+
+Переименованы следующие методы:
+
+- `findBlockInside()` в `findChildBlock()`
```

Comment:
and how to handle this part: https://github.com/bem/bem-components/blob/v3/common.blocks/popup/_target/popup_target_anchor.js#L84 ?

BTW, notice about new signature of `$.fn.bem()` should be added to migration guide.